baus/subsidies.py

```diff
@@ -617,30 +643,33 @@ def calculate_jobs_housing_fees(policy, year, buildings,
                                                      metadata=metadata)
 
 
-@orca.step()
-def subsidized_office_developer(feasibility, coffer, acct_settings, year,
-                                add_extra_columns_func, buildings, summary):
+#@orca.step()
+def subsidized_office_developer(feasibility, coffer, formula, year,
+                                add_extra_columns_func, buildings,
+                                summary, coffer_acct_name):
 
     # get the total subsidy for subsidizing office
-    total_subsidy = coffer["vmt_com_acct"].\
+    total_subsidy = coffer[coffer_acct_name].\
         total_transactions_by_subacct("regional")
 
     # get the office feasibility frame and sort by profit per sqft
-    feasibility = feasibility.to_frame().loc[:, "office"]
+    
+    feasibility = feasibility.loc[:, "office"]
+
     feasibility = feasibility.dropna(subset=["max_profit"])
 
-    feasibility["pda_id"] = feasibility.pda
+    # add necessary columns for filters
+    policy = orca.get_injectable("policy")
+    scenario = orca.get_injectable("scenario")
 
-    if "alternate_geography_scenarios" in acct_settings["vmt_settings"] \
-            and orca.get_injectable("scenario") in \
-            acct_settings["vmt_settings"]["alternate_geography_scenarios"]:
-        formula = acct_settings["vmt_settings"]["alternate_buildings_filter"]
-    else:
-        formula = acct_settings["vmt_settings"]["receiving_buildings_filter"]
+    if scenario in policy["geographies_pba40_enable"]:
+        feasibility["pda_id"] = feasibility.pda_pba40
+    elif scenario in policy["geographies_db_enable"] or \
+        scenario in policy["geographies_fb_enable"]:
+        feasibility["pda_id"] = feasibility.pda_pba50
 
     # filter to receiving zone
-    feasibility = feasibility.\
-        query(formula)
+    feasibility = feasibility.query(formula)
```

Comment:
@yuqiww where is "formula" defined?

Reply:
It is defined here: https://github.com/BayAreaMetro/bayarea_urbansim/pull/244/files/8dece214fe04667710705fa32cfa7e9f51ccd256#diff-8e32919ed5dab8c6af82aeeb375599e2981a9ac4427a7132bf89d899c7499616R1221

Reply:
So basically I've changed `subsidized_office_develop()` from an `orca.step()` to a function, and it has to be called by `subsidized_office_developer_lump_sum_accts()` or `subsidized_office_developer_lump_vmt()`.

Reply:
Very cool